examples/goldilocks_models/find_goldilocks_models.cc

```diff
@@ -1567,6 +1594,27 @@ int findGoldilocksModels(int argc, char* argv[]) {
        << max_ground_incline << endl;
   cout << "turning rate ranges from " << min_turning_rate << " to "
        << max_turning_rate << endl;
+
+  /// How to restrict the number of samples is still under testing
+  /// For now, the range of ground incline and stride length will not change while
+  /// the number of them decreases.
+  if(!uniform_grid && restricted_sample_number){
+      if(N_sample_gi>2){
+          N_sample_gi = pow(N_sample_gi, 0.5) + 1;
+      }
+      if(N_sample_sl>2){
+          N_sample_sl = pow(N_sample_sl, 0.5) + 1;
+      }
+      if(N_sample_tr>2){
+          N_sample_tr = pow(N_sample_tr, 0.5) + 1;
+      }
```

Comment:
The code here seems to be based on my older code. I pushed a few commits last week which allow even numbers of samples per task dimension

Reply:
No, it is used to avoid the situation of `N_sample = 1 or 2 `. Under these two situations, the number of samples will not decrease using the equations.